x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/get_high_level_stats.js

```diff
@@ -210,7 +212,10 @@ export function fetchHighLevelStats(server, callCluster, clusterUuids, start, en
         start,
         end,
         type: `${product}_stats`,
-        filters: [ { terms: { cluster_uuid: clusterUuids } } ]
+        filters: [
+          { terms: { cluster_uuid: clusterUuids } },
+          ...(isKibanaIndex ? [{ exists: { field: 'kibana_stats.usage.index' } }] : []),
```

Comment:
Because of the template changes,  will this cause problems if we backport?  
This field won't be in the mapping of existing indices and so the stats query will return nothing - so if we were to backport to 6.7 and 7.0 this would potentially break telemetry on clusters with existing monitoring indices?

Reply:
I believe for we'll have to add some sort of check for older versions if we need to backport. if the returned hits are empty then request again without the `existing` filter? The drawback of this is that it is being hit every 10seconds and now it might start hitting twice every 10s for those older versions. 

i'm not sure if it is a good approach to use the version number inside the condition but it prevents this issue

Reply:
Could we co-opt an existing mapped field on backported versions to indicate that usage data is present?  It's not ideal, but it would address the collection frequency using the same solution remains the same

Reply:
Then the filter here could just check for either condition, in a single query.

Reply:
It won't be pretty but you could use a [`script` query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-script-query.html) that does a bit of parsing and compairson on the `kibana_stats.kibana.version` field, since that's mapped in older versions:

https://github.com/elastic/elasticsearch/blob/3e60785faccbcf0ac766ec4f24acb32c260e948a/x-pack/plugin/core/src/main/resources/monitoring-kibana.json#L67-L69

Reply:
In fact, if we are going down the `script` query route, we could just check the existence of the `kibana_stats.usage` field with a `script` query.

Reply:
My [previous comment](https://github.com/elastic/kibana/pull/34609#discussion_r278246010) will not work. I chatted with @jdconrad off PR at length and learnt that we can't check the existence of `kibana_stats.usage` even with a `script` query since it's no indexed. This used to be possible pre-6.1 but hasn't been since.

So I think we could consider the approach I mentioned [two comments ago](https://github.com/elastic/kibana/pull/34609#discussion_r278243336), using the `kibana_stats.kibana.version` field.

Reply:
`keyword` fields support `range` queries, which we can use here to avoid a script and avoid checking for legacy versions, then re-submitting. We could change this code around to optionally add a more complex query that imitates the legacy code above in one pass:

```
{
  bool: {
    should: [
      { exists: { field: 'kibana_stats.usage.index' } },
      {
        bool: {
          should: [
            { range: { kibana_stats.kibana.version: { lt: '6.7.3' } } },
            { term: { kibana_stats.kibana.version: '7.0.0' } }
          ]
        }
      }
    ]
  }
}
```

Logically, this is then:

```
WHERE
  cluster_uuid IN clusterUuids
AND
  (
    kibana_stats.usage.index IS NOT NULL
  OR
    (
      kibana_stats.kibana.version < 6.7.3
    OR
      kibana_stats.kibana.version = 7.0.0
    )
  )
```

This would allow us to avoid the entire `someResultsAreLegacy` check because everything would be correct, for Kibana, with just this query, as well as avoid the `isLegacyVersion` check code altogether.

If `kibana_stats.usage.index` is enough to identify the with-usage documents, then I would avoid adding the `includes-usage-data` code also (https://github.com/elastic/kibana/pull/34609/files#diff-2a18558cd89400e7ad8801be1cb48e5cR60).

Reply:
@pickypg good point, I think this works provided we merge the updated template back to 6.7 or 7.0.  How often do we roll over the monitoring indices?  Since existing ones won't have `kibana_stats.usage.index` mapped, and so the query will return no usage stats.  If their daily indices that's probably OK, but if they're longer lived we might lose some usage telemetry.

Reply:
The monitoring index gets rolled daily. So we're in good shape either way, ironically, but with outdated info.

Imagine the user is running 6.7.0 - 6.7.2 without the fix, ingesting usage data with every payload. Then they upgrade to 6.7.3 midday, so the template doesn't have any impact until midnight UTC.

With the above query, we'd just grab the older version of its usage payload, which definitely falls in the "close enough" pile. Then, after tomorrow rolls around, the next time we'd get the new version.

> I think this works provided we merge the updated template back to 6.7 or 7.0.

Coincidentally I'm in the process of backporting the template to 6.7, so this should work out.

Reply:
BTW, a lot of the kudos go to @ycombinator for calling out the version script, which got me thinking about the `range` query with the version.

Reply:
OK looks like the default is to use daily indices, so I think we're good.